src/Form/inputTypeConfig.js

```diff
@@ -33,17 +33,19 @@ const inputTypeConfig = {
         type: 'credentials',
         getIconBase: () => ddgPasswordIcons.ddgPasswordIconBase,
         getIconFilled: () => ddgPasswordIcons.ddgPasswordIconFilled,
-        shouldDecorate: (input, {isLogin, device}) => {
+        shouldDecorate: (input, {isLogin, isSignup, device}) => {
             // if we are on a 'login' page, continue to use old logic, eg: just checking if there's a
             // saved password
             if (isLogin) {
                 return device.hasLocalCredentials
             }
 
             // at this point, it's not a 'login' attempt, so we could offer to provide a password?
-            const subtype = getInputSubtype(input)
-            if (subtype === 'password') {
-                return true
+            if (device.supportsFeature('password.generation') && isSignup) {
```

Comment:
`isLogin` and `isSignup` are mutually exclusive. Given that we return early with `isLogin`, this check is redundant.